x/twap/logic.go

```diff
@@ -10,13 +10,13 @@ import (
 	"github.com/osmosis-labs/osmosis/v12/x/twap/types"
 )
 
-func newTwapRecord(k types.AmmInterface, ctx sdk.Context, poolId uint64, denom0, denom1 string) (types.TwapRecord, error) {
+func newTwapRecord(k types.AmmInterface, keeper Keeper, ctx sdk.Context, poolId uint64, denom0, denom1 string) (types.TwapRecord, error) {
```

Comment:
In this case I suggest we just make this a keeper method instead of passing keeper as a parameter